File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/di/InjectionUtil.java
package pl.edu.amu.wmi.erykandroidcommon.di;

import android.content.Context;

/**
 * @author Eryk Mariankowski <eryk.mariankowski@247.codes> on 21.10.17.
 */
public class InjectionUtil {

    CommonApplicationComponent getCommonComponent(Context context) {
        return ((CommonApplication) context.getApplicationContext())
                .commonApplicationComponent();
    }

}

File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/di/InjectionUtil.kt
package pl.edu.amu.wmi.erykandroidcommon.di

import android.content.Context

/**
 * @author Eryk Mariankowski <eryk.mariankowski></eryk.mariankowski>@247.codes> on 21.10.17.
 */
class InjectionUtil {

    internal fun getCommonComponent(context: Context): CommonApplicationComponent? {
        return (context.applicationContext as CommonApplication)
                .commonApplicationComponent()
    }

}
